### ISSUE TITLE
Failure processing application bundle

### ISSUE THREAD

**Issue by neferin12** [neferin12/muse-sounds-manager-aur#1]
Error on start:
```
muse-sounds-manager 
Failure processing application bundle; possible file corruption. 
Arithmetic overflow while reading bundle. 
A fatal error occured while processing application bundle
```

**Comment by apapamarkou** [comment]
Confirm.  I try to run it on Arch under Hyprland and get the same message: 
```
Failure processing application bundle; possible file corruption.
Arithmetic overflow while reading bundle.
A fatal error occured while processing application bundle
```

I tried to replace the binaries from the .deb version but no luck.

**Comment by apapamarkou** [comment]
In Garuda KDE Lite works as expected. ver. 1.1.0.587 (x64)

**Comment by neferin12** [comment]
Sorry for the long delay, I'm currently having exams...; i was able to reproduce the problem in a minimal arch install (with gnome as DE). After some trying however, i was not able to get it running. Interestingly though, it also does work on both my normal machines that use KDE, which leads me to believe that there might be a dependency missing (although namcap shows no sign of that being the case). if anyone has any ideas I would be very grateful, otherwise I will poke around some more and maybe open an ticket on the support platform for the muse projects

**Comment by darioandreella** [comment]
Hey guys!

Not sure if it is because of MuseScore 4.2.1 or what else, but it's some 2 weeks that MuseScore crashes instantly **if attempting to play even one single note with muse-sounds**. I experience it in all my computers (most of them running Arch and KDE 6, the same happens with Manjaro and KDE 5.27.11 so no, not a KDE 6 issue as I thought initially).

I didn't have time to go deeper yet, but I find it strange that no one else is mentioning the problem, I tried it in 5 different machines, they behave all exactly the same (and they all worked before).

I tried also to remove muse-sound-manager-bin, check if I had the dependencies (that I in fact didn't, but so why did it work before?) and after re-compiling with no errors, it doesn't even open at all (Muse Sounds), while Muse Score keeps crashing.

Happy to contribute further.
Cheers!!

**Comment by neferin12** [comment]
Hi @darioandreella, thanks for your report!
The error with musescore crashing when using ms sounds is probably this issue with musescore itself: https://github.com/musescore/MuseScore/issues/21838

Also as far as I know there is no git version of the muse-sounds-manager, at least none that i maintain. Do you mean musescore-git or muse-sound-manager-bin?

**Comment by darioandreella** [comment]
> The error with musescore crashing when using ms sounds is probably this issue with musescore itself: [musescore/MuseScore#21838](https://github.com/musescore/MuseScore/issues/21838)

Hey again, thanks for the reply.
The issue described at the linke you provided is talking about MuseScore crashing in general. What I experience is, MuseScore crashing when I try to play something or drop down a note with ms-sounds. If I select the standard sounds, MuseScore works (quite fine).

Also, muse-sound-manager-bin itself (yes, sorry, I wrote -git in my previous message) doesn't even open at all now, even newly installed on a fresh Arch Linux (with KDE Plasma 6.0.3, kernel 6.8.4).

Cheers
Dario

**Comment by rasmusq** [comment]
I tried to run it on my desktop where I installed arch with the minimal configuration in archinstall and I get the same error message.
However, when I run it on my laptop where I installed arch with the gnome desktop using archinstall and installed Hyprland afterwards, it runs with no problem.

There might be some dependency that comes with the gnome desktop that simply isn't there in the minimal arch installation.

**Comment by jagjordi** [comment]
I have an arch installation with hyperland, I tried to install gnome and gnome-extras as @rasmusq but it did not solve the problem. Is there something else that can be done to debug this?

**Comment by rasmusq** [comment]
Yeah, i also realized that my reasoning was a bit off, as the one where I have gnome installed has had a ton of other stuff installed over the past 6 months. There could be a ton of things that one system supplies that the other one does not :/

**Comment by irvng-wav** [comment]
I experienced the exact same problem. Installed [muse-sounds-manager-bin<sup>aur</sup>](https://aur.archlinux.org/packages/muse-sounds-manager-bin) but the application didn't start when opening it from GNOME's app grid, and when opening it from the terminal it showed me the error mentioned in the first post.

I solved this by downloading the official `Muse Sounds Manager (DEB Beta)` from [Musescore's download page](https://musescore.org/en/download) and decompressing it with [dpkg](https://archlinux.org/packages/extra/x86_64/dpkg/) to the root partition. It looked like this: `sudo dpkg-deb -xv Muse_Sounds_Manager_Beta.deb /`

Definitely not ideal but now Muse Sounds Manager works great.

Additionally, Musescore installed from the official arch repos was crashing whenever I opened any score, as mentioned here:

> Hi @darioandreella, thanks for your report! The error with musescore crashing when using ms sounds is probably this issue with musescore itself: [musescore/MuseScore#21838](https://github.com/musescore/MuseScore/issues/21838)

To fix this I downloaded Musescore's AppImage from it's official download page and integrated it with [Gear Lever](https://flathub.org/apps/it.mijorus.gearlever) so it appeared in GNOME's app grid, then I uninstalled the problematic arch package.

That way I have Musescore and Muse Sounds Manager fully working together in Arch. I hope this helps someone!

**Comment by darioandreella** [comment]
Thanks @irvng-wav for the idea!

At first I used the Flatpak version of MuseScore just to carry on, because I needed to write stuff and get the job done, but it didn't see the Muse Sounds, they simply didn't show up in the mixer, so I still have no idea about if it works with Muse Sounds or not. It had also some visual quirks (the file manager was not themed ans so on) but so far it worked.

Now I tried the AppImage version as you suggested, was just missing fuse... and so far it works also with Muse Sounds and didn't crash!

I saw an update of Muse Sounds too, so it seems something has been indeed done also on that side, that was in fact the original topic of this discussion.

Now it seems to work, but I tried it only on one machine where it was already present, not a new install. I'll have the possibility to test with different machines later today.

The main problem is the version of Muse Score distributed through the Arch repo, that is the natural way to go for a normal user... it just still crashes with Muse Sounds and it also has problems with the menus, always showing up in the middle of the screen and not where they should be (included the sounds from the mixer), at least with Plasma 6.

Cheers, thanks for the useful discussion, hope we can get things to work together properly, it's a lovely suite and I look forward to have the good sounds back in place!

FOLLOW-UP: I tried on an other machine where only muse-sounds-manager was installed (and never the .deb version), and it still doesn't open. After double-checking that all the dependencies declaired in the [AUR page](https://aur.archlinux.org/packages/muse-sounds-manager-bin) are installed, and they are, launching it from the terminal I get "Failure processing application bundle; possible file corruption.
Arithmetic overflow while reading bundle.
A fatal error occured while processing application bundl"
